pennylane/interfaces/jax_jit.py

```diff
@@ -29,11 +29,17 @@
 Zero = jax.custom_derivatives.SymbolicZero
 
 
-def _numeric_type_to_dtype(numeric_type):
+def _numeric_type_to_dtype(numeric_type, device):
     """Auxiliary function for converting from Python numeric types to JAX
     dtypes based on the precision defined for the interface."""
 
     single_precision = dtype is jnp.float32
+    if numeric_type is bool:
+        if isinstance(device, qml.Device):
+            numeric_type = int
+        else:
+            return jnp.bool_

```

Comment:
This will go away if we start sampling in `int` instead of `bool`.

Reply:
done